src/trunk.c

```diff
@@ -7092,23 +7229,57 @@ trunk_range(trunk_handle  *spl,
    return rc;
 }
 
+/*
+ * Allocate memory for stats-related structures off the trunk.
+ */
+static trunk_stats *
+trunk_stats_init(trunk_handle *spl)
```

Comment:
Modularized common code from `trunk_create` and `trunk_mount`, so that the memory-related changes could be applied to just one place.